packages/starlight/utils/user-config.ts

```diff
@@ -271,6 +271,13 @@ const UserConfigSchema = z.object({
 
 	/** The default favicon for your site which should be a path to an image in the `public/` directory. */
 	favicon: FaviconSchema(),
+
+	/** Will be used as title delimiter in the generated `<title>` tag. */
+	titleDelimiter: z
+		.string()
+		.optional()
```

Comment:
We can use Zod’s `default()` method here to set the default delimiter. Also suggesting we use a `|` pipe character instead of `-` to match Docusaurus:

```suggestion
		.default('|')
```